package/MDAnalysis/lib/_cutil.pyx

```diff
@@ -49,6 +49,119 @@ ctypedef cset[int] intset
 ctypedef cmap[int, intset] intmap
 
 
+@cython.boundscheck(False) # turn off bounds-checking for entire function 
+@cython.wraparound(False)  # turn off negative index wrapping for entire function 
+cdef void _inverse_unique_unsorted_contiguous_array_inplace(
+						np.intp_t[::1] full_arr,
+                				np.intp_t[::1] unique_arr,
+                                		np.intp_t[::1] mask
+						):
+    """Find the inverse of a unique 1D array to map onto a non-unique 1D array.
+
+    Cython driver function. Loops though the unique array and finds the indices 
+    in the full array that correspond to the values in the unique array.
+
+    ie: np.array([1,2,3])[np.array(0,0,1,2,2)] == np.array([1,1,2,3,3])
+	       unique_arr[mask]                ==       full_arr
+
+
+    Calculates the bounding region of unsorted values to reduce number of iterations,
+    more efficient on uninterleaved full_arr, where unique_arr is sorted according to
+    first occurance of the value in full_arr. 
```

Comment:
I think this means `unique_arr` is *unsorted* and *ordered* by first appearance. The bounding region and efficiency note also is a bit of an implementation detail and should appear last, after the requirement of contiguous arrays, which is much more important.

Also, occurance -> occurrence